third_party/move/move-compiler-v2/tests/testsuite.rs

```diff
@@ -662,10 +669,10 @@ fn run_test(path: &Path, config: TestConfig) -> datatest_stable::Result<()> {
             test_output
                 .borrow_mut()
                 .push_str(&String::from_utf8_lossy(&out.into_inner()));
-            ok = check_diags(&mut test_output.borrow_mut(), &env);
+            ok = check_diags(&mut test_output.borrow_mut(), &env, &mut saw_warnings);
```

Comment:
Why this complexity here? Can't we just whenever we emit a diag on Warning level, actually emit it as Error when this flag is on? This PR seems way to complex to achieve the issue, with making the existing logic unnecessary complex. Or what am I missing?

Reply:
This mirrors the compiler itself which only exits on warnings at the end.  With the mixture of V1 (parser to expansion) and V2 (model, etc.) warnings it was especially tricky to promote warnings to errors.  It seemed easier, and probably more useful, to let warnings act like "nonblocking errors" and only have them halt compilation at the end.

(Btw, I think I answered this before, but may have lost it in the browser.)

Reply:
Is this a v1 problem? Can we then just isolate it for v1? I'm very concerned we increasing the complexity of the v2 compiler. I think even better to _only_ support in v2, as we are just a few weeks away from making it the default. In v2, you should be just able to go to the model and make a local change to add_diag or such.

Reply:
The v2 diagnostics don't even have nonblocking errors, so the resulting messages will be less useful.  First, this has nothing to do with the compiler; this is testing code.  This is simpler than trying to modify the error messages as they are generated; instead, we just modify the way we stop the compiler if errors are generated; now we also stop if warnings were generated.

Reply:
But we are calling the compiler in many places: the v2 driver, the test drivers (many of them), the prover, the formatter, the linter, ..

I prefer we do this so this client code does not break. Then this PR would become much simpler

- Just add a config option to ModelConfig warning-as-error
- Whenever a diag warning is submitted, turn it into `error: (from warning) <original message>` if that option is set. (I.e. prepend 'from warning' to the error message) 

No changes anywhere else, at least I think so.